### ISSUE TITLE
Liberty create default CN=localhost value for the key.p12 certificate file , Not the hostname where the server is running

### ISSUE THREAD

**Issue by YuLimin** [OpenLiberty/open-liberty#26083]
https://www.ibm.com/docs/en/was-liberty/nd?topic=jmx-configuring-secure-connection-liberty

About JConsole connect to Remote Liberty, here are shell on Windows:

```
set JAVA_HOME=C:\IBM\WebSphere\AppServer\9.0.5\java\8.0
set WLP_HOME=C:\IBM\WebSphere\Liberty\wlp-nd-all-23.0.0.7
%JAVA_HOME%\bin\jconsole -J-Djava.class.path=%JAVA_HOME%\lib\jconsole.jar;%JAVA_HOME%\lib\tools.jar;%WLP_HOME%\clients\restConnector.jar -J-Djavax.net.ssl.trustStore=<PATH>\key.p12 -J-Djavax.net.ssl.trustStorePassword=P@ssw0rd -J-Djavax.net.ssl.trustStoreType=PKCS12
```

But can use JConsole connect Liberty by Local Process, and can get mbeans information from browser via URL /IBMJMXConnectorREST/mbeans/
But can NOT connect Liberty by Remote Process.
service:jmx:rest://yuliminp15:9443/IBMJMXConnectorREST

Error is : Secure connection failed. Retry insecurely?
but still got Error : Connection Failed. Retry?

and can NOT open key.p12 file via ikeyman tool. will got error : Error extracting keyentry aliases from PFX

server.xml
```
<?xml version="1.0" encoding="UTF-8"?>
<server description="JMX Remote">
	<!-- Enable features -->
	<featureManager>
		<feature>jsp-2.3</feature>
		<feature>restConnector-2.0</feature>
		<feature>transportSecurity-1.0</feature>
	</featureManager>
	<!-- To access this server from a remote client add a host attribute to 
		the following element, e.g. host="*" -->
	<httpEndpoint id="defaultHttpEndpoint" httpPort="9080" httpsPort="9443" host="*" />
	<!-- Automatically expand WAR files and EAR files -->
	<applicationManager autoExpand="true" />
	<administrator-role>
		<user>admin</user>
	</administrator-role>
	<basicRegistry id="basic">
		<user name="admin" password="{aes}APBXmPlx9ilC3j3MTHEC0u8i2M1VLCuvwxj3gFUXMlhu" />
	</basicRegistry>
	<!-- inbound (HTTPS) keystore -->
	<keyStore id="defaultKeyStore" password="{xor}Dx8sLChvLTs=" location="${server.config.dir}/resources/security/key.p12" />
</server>
```

iKeyman tool can't open the default key store file key.p12, will cause Error extracting keyentry aliases from PFX.
but can run openssl shell to get certificate information:
openssl pkcs12 -in key.p12 -passin pass:P@ssw0rd -passout pass:P@ssw0rd

I found the default CN value of the key.p12 certificate file is CN=**localhost**, Not the hostname of the machine where the server is running. such as CN=**libertyhost1**

then I create a new key store file key.p12 by iKeyman manually, create a self signed certificate and set CN value as the hostname of the machine(**libertyhost1**). and set location value in the server.xml
<keyStore id="defaultKeyStore" password="P@ssw0rd" location="/IBM/WebSphere/Liberty/WAS_JDK_Liberty_Key.p12" />

then start Liberty and JConsole can connect Liberty JMX by Remote Process and solve this issue.

Maybe Liberty have issue about create the CN default value during set <keyStore id="defaultKeyStore".../> in order to create the key store file, should be set the hostname of the machine where the server is running instead of set the default value as **localhost**.

### 
The same issue on JDK 8.0 shipped with WAS and IBM Semeru OpenJDK on **Windows and Linux**

`**java -version**`
java version "1.8.0_331"
Java(TM) SE Runtime Environment (build 8.0.7.10 - pwa6480sr7fp10-20220505_01(SR7 FP10))
IBM J9 VM (build 2.9, JRE 1.8.0 Windows 11 amd64-64-Bit Compressed References 20220427_27745 (JIT enabled, AOT enabled)
OpenJ9   - b15041a
OMR      - 3671a9f
IBM      - 1b0232b)
JCL - 20220504_01 based on Oracle jdk8u331-b09

`**java -version**`
openjdk version "17.0.7" 2023-04-18
IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7)
Eclipse OpenJ9 VM 17.0.7.0 (build openj9-0.38.0, JRE 17 Windows 11 amd64-64-Bit Compressed References 20230418_439 (JIT enabled, AOT enabled)
OpenJ9   - d57d05932
OMR      - 855813495
JCL      - 9d7a231edbc based on jdk-17.0.7+7)

Liberty version : 23.0.0.7

**Comment by una-tapa** [comment]
Summary:
Liberty creating a cert with "CN=localhost".  The workaround from Li Min is to recreate it with "CN=actual_host_name"

**Comment by adriankalafut** [comment]
@YuLimin 
The **CN** can be changed to your preferred host name (i.e. **libertyhost1**) by adding the following tag to your server.xml file:
```<variable name="defaultHostName" value="libertyhost1" />```
where the **value** attribute will reflect what is stored in the CN.

**Comment by adriankalafut** [comment]
@YuLimin
From my end, I didn't have any issues with iKeyman tool opening the default key store (i.e. key.p12). 
If you can provide us the 'bad' key.p12 file, we can take a closer look on what the issue may be.

**Comment by adriankalafut** [comment]
Generating a 'bad' key.p12 file (which could not be opened with iKeyman Tool) is not reproducible from our end.
From our end, we tested the following use-cases in attempt to reproduce the issue (in a Windows OS env):
- Generate key.p12 file by Liberty running with IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7) using the same server.xml file provided in this issue.
- Generate key.p12 file by Liberty running with IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7) using the server.xml file and additional variable tag (i.e. defaultHostName) to change the CN value.
- Generate key.p12 file using iKeyman.

After each of these use-cases, we were able to successfully open the key.p12 generated using iKeyman, hence we did not receive the same error reported in this issue (i.e. Error extracting keyentry aliases from PFX). 
Additionally, we were provided a copy of the 'bad' key.p12 file by @YuLimin and were also able to open the file successfully using iKeyman tool in our Windows OS env. 
One thing I did notice is that the error occurred when transferring the key.p12 file from one OS platform to another (i.e. Windows to macOS). However, this error occurred for all key.p12 files generated by the use-cases above.

When creating the key.p12, we found that there are some inconsistencies between securityUtility and our runtime code. Specifically, the information contained in the SAN portion. Thus, we will create a PR fix these inconsistencies.

**Comment by YuLimin** [comment]
Additionally, we were provided a copy of the 'bad' key.p12 file by @YuLimin and were also able to open the file successfully using iKeyman tool in our Windows OS env. 

// Which version of you iKeyMan Tool? My ikeyman tool version is 8.0.424 shipped with WAS 9.0.5.12 and java version "1.8.0_331"
Java(TM) SE Runtime Environment (build 8.0.7.10 - pwa6480sr7fp10-20220505_01(SR7 FP10))
IBM J9 VM (build 2.9, JRE 1.8.0 Windows 11 amd64-64-Bit Compressed References 20220427_27745 (JIT enabled, AOT enabled)
OpenJ9   - b15041a
OMR      - 3671a9f
IBM      - 1b0232b)
JCL - 20220504_01 based on Oracle jdk8u331-b09

After add config to server.xml
`<variable name="defaultHostName" value="libertyhost1" />`
will generate CN value with the right hostname.

And I found  ikeyman tool shipped with WAS can't open the key.p12 file which generated by IBM Semeru Runtime Open Edition 17.0.7.0 (build 17.0.7+7) during Liberty starting.

But can open the key.p12 file which generated by WAS JDK 8.0 during Liberty starting.

**Comment by adriankalafut** [comment]
@YuLimin I confirm this issue occurs from my end as well when opening the key.p12 file using IKeyman tool version 8.0.424.
However, this issue does not occur with IKeyman tool version 8.0.425.

**Comment by adriankalafut** [comment]
For now, we have decided to leave the inconsistencies found between our securityUtility and run-time code. We will re-visit this issue in the future if necessary.

The underlying issues here have been resolved:
- Key.p12 file can be opened using a version of IKeyman tool version > 8.0.424.
- The CN can be changed to the preferred host name by adding the **defaultHostName** variable tag to the server.xml file.

Closing this issue.

**Comment by YuLimin** [comment]
Yes. I have double confirmed them:

1. key.p12 can be open by iKeyMan 8.0.425 shipped with WAS 9.0.5.17, java version is
```
java version "1.8.0_381"
Java(TM) SE Runtime Environment (build 8.0.8.11 - pwa6480sr8fp11-20230901_01(SR8 FP11))
IBM J9 VM (build 2.9, JRE 1.8.0 Windows 11 amd64-64-Bit Compressed References 20230817_56476 (JIT enabled, AOT enabled)
OpenJ9   - c425e1c
OMR      - e712c65
IBM      - 696e9df)
JCL - 20230811_01 based on Oracle jdk8u381-b09
```

2. CN name can be generated by adding the **defaultHostName** variable in the server.xml file.
`<variable name="defaultHostName" value="yulimin.abc-domain.net" />`

Thanks a lot.
